lib/vrl/compiler/src/value/convert.rs

```diff
@@ -153,359 +151,40 @@ impl VrlValueConvert for Value {
     }
 }
 
-// Value::Integer --------------------------------------------------------------
-
-impl Value {
-    pub fn is_integer(&self) -> bool {
-        matches!(self, Value::Integer(_))
-    }
-
-    pub fn as_integer(&self) -> Option<i64> {
-        match self {
-            Value::Integer(v) => Some(*v),
-            _ => None,
-        }
-    }
-}
-
-impl From<i8> for Value {
-    fn from(v: i8) -> Self {
-        Value::Integer(v as i64)
-    }
-}
-
-impl From<i16> for Value {
-    fn from(v: i16) -> Self {
-        Value::Integer(v as i64)
-    }
-}
```

Comment:
I noticed you removed these in favour of the macro impl. I'd love for these to replace the macro because rust-analyzer doesn't always play nice with macro's when using its features, and since these impls are a write-once thing, it's only 5 minutes of extra work and we never touch them again.

Reply:
I will gladly remove macros :)